### DISCUSSION TITLE
IKeyValueWatcher.Watch not called when entry deleted because of TTL

### DISCUSSION THREAD

**Discussion by simonxbto** [nats-io/nats.java#640] [General]
Hello,

The method IKeyValueWatcher.Watch is not called when an KeyValueEntry is delete because of TTL expiration.
I am using IKeyValue.WatchAll to subscribe.
Can you confirm this is expected behavior ?

Regards,
Simon

**Comment by scottf** [comment]
Yes, that's what I would expect if you started the watch after the TTL expiration. The TTL is accomplished by leveraging the max_age limit retention policy on a stream, and a key is really a specific subject of messages.

**Comment by scottf** [comment]
@simonxbto Should have added this note: If a key entry is deleted because of expiry, the watch will not get a notification of this removal. There is discussion with the server team on how to handle this. Consider this a known issue.

**Reply by simonxbto** [reply]
ok. I think there is the same issue on the dotnet client

**Reply by scottf** [reply]
@simonxbto This is not a client issue, this is a server issue, so it will happen on every client.
